_posts/2015-06-09-geomesa-kafka-quickstart.md

```diff
@@ -0,0 +1,128 @@
+---
+title: GeoMesa Kafka Quick Start
+author: atatusko
+layout: tutorial
+redirect_from:
+    - /2015/06/09/geomesa-kafka-quickstart/
+---
+
+{% include tutorial-header.html %}
+
+## Background
+
+[Apache Kafka](http://kafka.apache.org/) is "publish-subscribe messaging rethought as a distributed commit log." In the context of GeoMesa, Kafka is a useful tool for working with streams of geospatial data. Interaction with Kafka in GeoMesa occurs with the KafkaDataStore which implements the [GeoTools DataStore interface](http://docs.geotools.org/latest/userguide/library/data/datastore.html).
+
+This quickstart tutorial is bundled as a Java program which will introduce how to produce and consume messages in Kafka using GeoMesa. The tutorial will also show how to query the data and replay the messages in a Kafka topic to achieve an earlier state. The tutorial uses GeoServer as a quick way to visualize the changes being made in Kafka.
+
+## Prerequisites
+
+* basic knowledge of [GeoTools](http://www.geotools.org), [GeoServer](http://geoser!!ver.org), and Kafka
+* access to a Kafka 0.8.2.x server with an appropriate Zookeeper instance(s)
+* access to GeoServer version 2.5.2
+* a local copy of the Java Development Kit 1.7.x
+* Apache Maven installed
+* a GitHub client installed
+
+## Setup
+
+You must go through the [GeoMesa Deployment tutorial](http://geomesa.org/geomesa-deployment/) first, completing the tasks relevant to Kafka. The deployment tutorial prioritizes Accumulo deployment so scroll down for the Kafka deployment section.
+
+## Run the code
+
+Ensure your Kafka and Zookeeper instances are running. You can use [Kafka's quickstart](http://kafka.apache.org/documentation.html#quickstart) to get Kafka/Zookeeper instances up and running quickly.
+
+Navigate to your geomesa directory. On the command-line run the quickstart program:
+
+{% highlight bash %}
+java -cp ./geomesa-examples/kafka-quickstart/target/kafka-quickstart-{{ site.developmentVersion }}.jar org.locationtech.geomesa.examples.KafkaQuickStart -brokers "localhost:9092" -zookeepers "localhost:2181"
+{% endhighlight %}
+
+where you provide your own values for the following arguments:
+
+* ```brokers```: your Kafka broker instances, seperated by commas.
+* ```zookeepers```: your Zookeeper nodes, separated by commas.
+* ```zkPath```: Zookeeper's path where metadata is stored. Defaults to /geomesa/ds/kafka.
+
+The program will create some metadata in zookeeper and an associated topic in your Kafka instance and pause execution to let you add the newly created KafkaDataStore to GeoServer.
+
+## Visualize with GeoServer
+
+#### Register the GeoMesa store with GeoServer
+
+Log into GeoServer using your user and password credentials. Click “Stores” in the left-hand gutter and “Add new Store”. If you do not see the Kafka Data Store listed under Vector Data Sources, ensure the plugin and dependencies are in the right directory and restart GeoServer.
```

Comment:
I don't know that it matters in this context, but it appears as though your editor has inserted smart quotes.